content/blog/2019-12-18-qvec/index.md

```diff
@@ -0,0 +1,145 @@
+
+
+## Quantum Computing
+
+Quantum Computing has exploded into the popular imagination in the past decade due to the promise of massive theoretical speedups over conventional digital computers. Whether real quantum hardware can live up to the promise remains to be seen, but that has not stopped researchers from developing complex toolflows and algorithms.
+
+The computing paradigm of quantum computers is inherently different from a standard "classical" computer. Instead of representing a bit as either a `0` or `1`, quantum bits (qubits) represent a bit of information using a quantum superposition `a |0> + b |1>`, where `|0>` and `|1>` represent possible realizable states and `a` and `b` are normalized constants related to probability of measuring the respective state. Although `a` and `b` theoretically hold infinite information, it is only practically possible to measure a bit of information from the state as in the classical case as the state collapses to one of the realizable state `|0>` or `|1>` upon measurement.
+
+Quantum computing offers unique computing properties due to the nature of a qubit. The main computational differences between quantum and classical computing include the following properties:
+
+|  Property       |  Quantum    | Classical CPU |
+| ------  | ------------- | ------------------- |
+| Architecture   | Spatial | Von-Neumann  |
+| Data    | Quantum State | Voltage Level |
+| Control | External (Laser) | Voltage Level |
+| States per bit | Exponential | Linear |
+
+In general, quantum computer can implement all of the computation primitives that a classical one can. Both, in theory, can be turing complete with a universal set of logic gates. However, quantum computing also has computational primitives that classical computers don't share. These primitives are key to quantum supremacy: the concept that a quantum computer can theoretically outperform classical computers.
+
+| Unique compute | Example Usage |
+| ----------------| ------------- |
+| Large State Space  | Chemical Reaction Simulation |
+| Entanglement | Combinatorial Optimization (TSP) |
+| Amplitude Magnification | Database Search |
+| Probabilistic (multiple results) | ? |
+| Phase | ? |
+
+A potential downside to quantum computing is that it is inherently probabilistic. An output on one execution may be entirely different from the output on the next run. Quantum algorithms must be designed so that the correct answer must have measurement probability >50%. The answer can then be inferred by repeating the execution many times and taking the majority result. Many quantum algorithms exist in Bounded Quantum polynomial class (BQP) where the correct answer can be found in polynomial time with probability at least 2/3.  It can, however, practically be time and resource intensive to run quantum programs a sufficient number of times to achieve a reasonable confidence.
```

Comment:
```suggestion
A potential downside to quantum computing is that it is inherently probabilistic. An output on one execution may be entirely different from the output on the next run. Quantum algorithms must be designed so that the correct answer must have measurement probability >50%. The answer can then be inferred by repeating the execution many times and taking the majority result. Many quantum algorithms exist in Bounded Quantum Polynomial class (BQP) where the correct answer can be found in polynomial time with probability at least 2/3.  It can, however, practically be time and resource intensive to run quantum programs a sufficient number of times to achieve a reasonable confidence.
```